packages/web-frontend/src/containers/Map/Map.js

```diff
@@ -0,0 +1,289 @@
+import { checkBoundsDifference, organisationUnitIsArea } from '../../utils';
+import { DemoLand } from './DemoLand';
+import { ConnectedPolygon } from './ConnectedPolygon';
+import { DisasterLayer } from './DisasterLayer';
+import { ZoomControl } from './ZoomControl';
+import {
+  selectActiveTileSet,
+  selectAllMeasuresWithDisplayInfo,
+  selectCurrentOrgUnit,
+  selectHasPolygonMeasure,
+  selectOrgUnit,
+  selectOrgUnitChildren,
+  selectOrgUnitSiblings,
+  selectRenderedMeasuresWithDisplayInfo,
+} from '../../selectors';
+import { changePosition, closeDropdownOverlays, setOrgUnit } from '../../actions';
+import { TRANS_BLACK, TRANS_BLACK_LESS } from '../../styles';
+
+const CHANGE_TO_PARENT_PERCENTAGE = 0.6;
+
+const StyledMap = styled(LeafletMap)`
+  height: 100vh;
+  width: 100%;
+
+  .leaflet-control-zoom {
+    z-index: 1;
+    border: none;
+    top: -50px;
+    right: 350px;
+    transition: right 0.5s ease;
+
+    a {
+      background: ${TRANS_BLACK_LESS};
+      box-shadow: none;
+      border: none;
+      color: white;
+
+      &:hover {
+        background: ${TRANS_BLACK};
+        box-shadow: none;
+      }
+    }
+  }
+`;
+
+/**
+ * Map
+ * Includes basic map setup/rendering, controlled through props that are connected to the redux store.
+ * Rendering includes heatmaps markers, polygons.
+ */
+class MapComponent extends Component {
+  componentWillMount() {
+    window.addEventListener('resize', () => this.forceUpdate());
+  }
+
+  shouldComponentUpdate(nextProps) {
+    const {
+      currentOrganisationUnit,
+      displayedChildren,
+      measureInfo,
+      measureData,
+      position,
+      tileSetUrl,
+      sidePanelWidth,
+    } = this.props;
+    // Only updates/re-renders when the measure has changed or the orgUnit has changed.
+    // These are the only cases where polygons or area tooltips should rerender.
+    if (nextProps.measureInfo.measureId !== measureInfo.measureId) return true;
+
+    if (JSON.stringify(nextProps.displayedChildren) !== JSON.stringify(displayedChildren))
+      return true;
+
+    if (
+      (nextProps.currentOrganisationUnit || {}).organisationUnitCode !==
+      (currentOrganisationUnit || {}).organisationUnitCode
+    ) {
+      return true;
+    }
+
+    if (nextProps.tileSetUrl !== tileSetUrl) return true;
+
+    if (nextProps.sidePanelWidth !== sidePanelWidth) return true;
+
+    if (JSON.stringify(nextProps.position) !== JSON.stringify(position)) return true;
+
+    if (JSON.stringify(nextProps.measureData) !== JSON.stringify(measureData)) return true;
+
+    return false;
+  }
+
+  onPositionChanged = (center, bounds, zoom) => {
+    const { position, onChangePosition } = this.props;
+
+    // only check when zooming _out_
+    if (zoom < position.zoom) {
+      this.checkZoomOutToParentOrgUnit(bounds);
+    }
+
+    // Notify redux that we've moved
+    onChangePosition(center, zoom);
+  };
+
+  checkZoomOutToParentOrgUnit(bounds) {
```

Comment:
Worth noting that this didn't make it into ui-components, so zoom out to parent won't be available on other platforms. I think that's fine for now

Reply:
yes that is an interesting point. I guess I didn't move it down into the Leaflet map component because it seemed like a step change in awareness for the ui-components map. It would be aware of the map parent and orgunits suddenly where it isn't at the moment. It may be that it's worth making it aware of them and it wouldn't be difficult to shift it down though so I don't feel strongly about it either way. Perhaps keep it as is for now and move it later if it makes sense to.

Reply:
Yep sure, keep as is for now and move later if other platforms want to start doing that